tests/Http/RequestTest.php

```diff
@@ -70,6 +70,25 @@ public function testData()
 		$this->assertNull($request->get('c'));
 	}
 
+	public function testDataNumeric()
+	{
+		$request = new Request([
+			'body'   => [1 => 'a'],
+			'query'  => [
+				'0' => 'b',
+				2   => 'c'
+			]
+		]);
+
+		$this->assertSame([1 => 'a', 0 => 'b', 2 => 'c'], $request->data());
```

Comment:
I think in the real world, all keys in the `body` and `query` input arrays should be numeric strings instead of integers (but would be good to verify that).

However with numeric strings, the call to `$request->data()` should also return numeric strings as keys and not convert them to ints. In this test `'0'` starts out as a numeric string but suddenly seems to get converted to an integer, which is IMO wrong. Why is that happening?

I'd expect the following result:

```suggestion
		$this->assertSame(['1' => 'a', '0' => 'b', '2' => 'c'], $request->data());
```

Reply:
Dump for `?1=b&0=a&2=c&foo=bar`

### Output
````php
// var_dump($_GET);
array(4) {
  [1]=>
  string(1) "b"
  [0]=>
  string(1) "a"
  [2]=>
  string(1) "c"
  ["foo"]=>
  string(3) "bar"
}

// var_dump(get());
array(4) {
  [1]=>
  string(1) "b"
  [0]=>
  string(1) "a"
  [2]=>
  string(1) "c"
  ["foo"]=>
  string(3) "bar"
}
````

So `get()` integer array index is same with `$_GET`

Reply:
@lukasbestle You will also be surprised! 😅 

````php
var_dump(['0' => 'Wow!']);

array(1) {
  [0]=>
  string(4) "Wow!"
}
````

Reply:
I'm indeed surprised by yet another PHP quirk I wasn't aware of until now. 🤯
PHP actually converts numeric keys to integers internally. Not just in the `var_dump()` output, but even when calling `is_int($key))` within a loop for example.